packages/mermaid/src/diagrams/block/blockDB.ts

```diff
@@ -0,0 +1,381 @@
+  cnt++;
+  return 'id-' + Math.random().toString(36).substr(2, 12) + '-' + cnt;
+};
+
+type ISetHierarchy = (block: Block[]) => void;
+const setHierarchy = (block: Block[]): void => {
+  rootBlock.children = block;
+  populateBlockDatabase(block, rootBlock);
+  blocks = rootBlock.children;
+};
+
+type IAddLink = (link: Link) => Link;
+const addLink: IAddLink = (link: Link): Link => {
+  links.push(link);
+  return link;
+};
+
+type IGetColumns = (blockid: string) => number;
+const getColumns = (blockid: string): number => {
+  const block = blockDatabase[blockid];
+  if (!block) {
+    return -1;
+  }
+  if (block.columns) {
+    return block.columns;
+  }
+  if (!block.children) {
+    return -1;
+  }
+  return block.children.length;
+};
+
+type IGetBlocks = () => Block[];
+/**
+ * Returns all the blocks as a flat array
+ * @returns
+ */
+const getBlocksFlat: IGetBlocks = () => {
+  const result: Block[] = [];
+  // log.debug('abc88 getBlocksFlat', blockDatabase);
+  const keys = Object.keys(blockDatabase);
+  for (const key of keys) {
+    result.push(blockDatabase[key]);
+  }
+  return result;
+};
+/**
+ * Returns the the hierarchy of blocks
+ * @returns
+ */
+const getBlocks: IGetBlocks = () => {
+  return blocks || [];
+};
+type IGetEdges = () => Block[];
+const getEdges: IGetEdges = () => {
+  return edgeList;
+};
+type IGetBlock = (id: string) => Block | undefined;
+const getBlock: IGetBlock = (id: string) => {
+  return blockDatabase[id];
+};
+type ISetBlock = (block: Block) => void;
+const setBlock: ISetBlock = (block: Block) => {
+  blockDatabase[block.id] = block;
+};
+
+type IGetLinks = () => Link[];
+const getLinks: IGetLinks = () => links;
+
+type IGetLogger = () => Console;
+const getLogger: IGetLogger = () => console;
+
+type IGetClasses = () => Record<string, ClassDef>;
+/**
+ * Return all of the style classes
+ */
+export const getClasses = function () {
+  return classes;
+};
+export interface BlockDB extends DiagramDB {
+  clear: () => void;
+  getConfig: () => BlockConfig | undefined;
+  addLink: IAddLink;
+  getLogger: IGetLogger;
+  getEdges: IGetEdges;
+  getBlocksFlat: IGetBlocks;
+  getBlocks: IGetBlocks;
+  getBlock: IGetBlock;
+  setBlock: ISetBlock;
+  getLinks: IGetLinks;
+  getColumns: IGetColumns;
+  getClasses: IGetClasses;
+  typeStr2Type: ITypeStr2Type;
+  edgeTypeStr2Type: IEdgeTypeStr2Type;
+  edgeStrToEdgeData: IEdgeStrToEdgeDataType;
+  setHierarchy: ISetHierarchy;
+  generateId: IGenerateId;
+}
+
+const db: BlockDB = {
+  getConfig: () => configApi.getConfig().block,
+  addLink: addLink,
+  typeStr2Type: typeStr2Type,
+  edgeTypeStr2Type: edgeTypeStr2Type,
+  edgeStrToEdgeData,
+  getLogger, // TODO: remove
+  getBlocksFlat,
+  getBlocks,
+  getEdges,
+  getLinks,
+  setHierarchy,
+  getBlock,
+  setBlock,
+  // getAccTitle,
+  // setAccTitle,
+  // getAccDescription,
+  // setAccDescription,
+  // getDiagramTitle,
+  // setDiagramTitle,
+  getColumns,
+  getClasses,
+  clear,
+  generateId,
+};
```

Comment:
```suggestion
const db = {
  getConfig: () => configApi.getConfig().block,
  addLink: addLink,
  typeStr2Type: typeStr2Type,
  edgeTypeStr2Type: edgeTypeStr2Type,
  edgeStrToEdgeData,
  getLogger, // TODO: remove
  getBlocksFlat,
  getBlocks,
  getEdges,
  getLinks,
  setHierarchy,
  getBlock,
  setBlock,
  getColumns,
  getClasses,
  clear,
  generateId,
} as const;

export type BlockDB = typeof db & DiagramDB;
```

This would help us remove all the `type IGenerateId = () => string;`

Reply:
Thx